lib/Sema/CSDiag.cpp

```diff
@@ -2477,6 +2477,14 @@ diagnoseUnviableLookupResults(MemberLookupResult &result, Type baseObjTy,
           diagnose(loc, diag::did_you_mean_raw_type);
           return; // Always prefer this over typo corrections.
         }
+      } else if (baseObjTy->isAny()) {
+        if (auto DRE = dyn_cast<DeclRefExpr>(baseExpr)) {
+          auto name = DRE->getDecl()->getName().str().str();
+          std::string replacement = "(" + name + " as AnyObject)";
```

Comment:
Also, why limit the note to DeclRefExprs? I imagine one of the most common cases of this is "drilling into a plist/JSON dictionary or array".